src/pages/iou/request/step/IOURequestStepScan/index.native.tsx

```diff
@@ -161,24 +179,64 @@ function IOURequestStepScan({
         Navigation.goBack();
     };
 
-    const navigateToConfirmationStep = useCallback(() => {
-        if (backTo) {
-            Navigation.goBack(backTo);
-            return;
-        }
+    const navigateToConfirmationStep = useCallback(
```

Comment:
Isn't the logic from all these navigates kinda duplicated? It feels you are doing almost the same in each

Reply:
Hmmm. I agree it is pretty similar, but we do have to differentiate between scan, manual, and distance flows, as they are different API calls. I could move it somewhere else to avoid the duplication getting participants for splits, but the rest of the code would remain, so I'm not sure how much worth is in that 🤷